### PR TITLE
BLD: pin extension-helpers to 1.* following upstream recommendation

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description

Following https://github.com/astropy/extension-helpers/pull/72

ping @hamogu and @pllim

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/changes/15682.other.rst
Comment (pllim) [inline docs/changes/15682.other.rst:1]: Does not need a change log. This is not something end-users would notice or care. Please delete the change log fragment and squash it out. Thanks!

Comment (neutrinoceros) [reply docs/changes/15682.other.rst:1]: I keep forgetting that's the rule, sorry ! and done !

Comment (pllim) [reply docs/changes/15682.other.rst:1]: No worries!

**Review by pllim — APPROVED** [review]
LGTM now. Thanks!

I think backporting would be acceptable since we're future proofing the release branch.

**Pull request merged**